File: ABC01_99/ABC33/c.cpp
#include <cstring>
#include <iostream>
#include <stack>
using namespace std;

int main() {
  string S;
  cin >> S;
  stack<long long int> n;
  long long int cur = S[0] - '0';
  n.push(cur);

  for (int i = 1; i < S.length(); i += 2) {
    if (S[i] == '+')
      n.push(S[i + 1] - '0');
    else {
      cur = n.top() * (S[i + 1] - '0');
      n.pop();
      n.push(cur);
    }
  }

  long long int ans = 0;
  while (!n.empty()) {
    cur = n.top();
    if (cur > 0) ans++;
    n.pop();
  }

  cout << ans << endl;
  return 0;
}

File: ABC01_99/ABC33/c.py
from collections import deque


def main():
    S = input()
    d = deque([])
    d.append(int(S[0]))
    for i in range(1, len(S)):
        if (i % 2):
            if (S[i] == '+'):
                d.append(int(S[i + 1]))
            else:
                cur = d[-1] * int(S[i + 1])
                d.pop()
                d.append(cur)
    ans = len(d)
    print(ans - d.count(0))


if __name__ == "__main__":
    main()
